File: affiliate_api/app/models/referral_link.py
# Source: AffiliateMarketing.Domain/Entities/ReferralLink.cs
import uuid
from datetime import datetime, timezone
from sqlalchemy import String, Boolean, DateTime, ForeignKey
from sqlalchemy.orm import mapped_column, Mapped, relationship
from app.core.db import Base

class ReferralLink(Base):
    __tablename__ = "referral_links"

    id: Mapped[uuid.UUID] = mapped_column(primary_key=True, default=uuid.uuid4)
    affiliate_profile_id: Mapped[uuid.UUID] = mapped_column(ForeignKey("affiliate_profiles.id"), nullable=False)
    product_id: Mapped[uuid.UUID] = mapped_column(ForeignKey("products.id"), nullable=False)
    generated_code: Mapped[str] = mapped_column(String, nullable=False, unique=True)
    source_platform: Mapped[str | None] = mapped_column(String, nullable=True)
    is_active: Mapped[bool] = mapped_column(Boolean, default=True)
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), default=lambda: datetime.now(timezone.utc))

File: AffiliateMarketing.Domain/Entities/ReferralLink.cs
namespace AffiliateMarketing.Domain.Entities;

public sealed class ReferralLink
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid AffiliateProfileId { get; set; }
    public Guid ProductId { get; set; }

    // The unique string in the URL (e.g., "startup-deal-jimmy")
    public string GeneratedCode { get; set; } = default!;

    // For Phase 7: Tracking if this link was for FB, IG, or YouTube
    public string? SourcePlatform { get; set; }

    public bool IsActive { get; set; } = true;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public AffiliateProfile AffiliateProfile { get; set; } = default!;
    public Product Product { get; set; } = default!;
}